vpr/src/device/rr_graph_builder.h

```diff
@@ -40,6 +40,37 @@ class RRGraphBuilder {
     t_rr_graph_storage& node_storage();
     /** @brief Return a writable object for update the fast look-up of rr_node */
     RRSpatialLookup& node_lookup();
+
+    /** @brief Add a rr_segment to the routing resource graph. Return an valid id if successful.*/
+    inline RRSegmentId add_rr_segment(const t_segment_inf& segment_info) {
+        //Allocate an ID
+        RRSegmentId segment_id = RRSegmentId(segment_ids_.size());
+        segment_ids_.push_back(segment_id);
+
+        rr_segments_.push_back(segment_info);
+
+        return segment_id;
+    }
+    /** @brief Return a writable list of all the rr_segments */
+    inline vtr::vector<RRSegmentId, t_segment_inf>& rr_segments() {
+        return rr_segments_;
+    }
+
+    /** @brief Add a rr_swtich_inf to the routing resource graph. Return an valid id if successful.*/
+    inline RRSwitchId add_rr_switch(const t_rr_switch_inf& switch_info) {
```

Comment:
As suggested by @vaughnbetz , we need to provide more intuitive comments to guide developers

```
/** @brief Add a rr_switch to the routing resource graph. Return an valid id if successful.
  *  Each rr_switch contains the detailed information of a routing switch interconnecting two routing resource nodes. It is frequently used by client functions in timing prediction. We use a fly-weight data structure here, in the same philosophy as the rr_indexed_data. See detailed explanation on rr_switch in the t_rr_switch_inf data structure
  */
```